wemake_python_styleguide/visitors/ast/attributes.py

```diff
@@ -73,3 +74,38 @@ def _check_magic_attribute(self, node: ast.Attribute) -> None:
                 self._ensure_attribute_type(
                     node, DirectMagicAttributeAccessViolation,
                 )
+
+
+@final
+class EncodingVisitor(BaseNodeVisitor):
+    """Check if open function has the encoding parameter."""
+
+    def __init__(self, *args, **kwargs) -> None:
+        """Creates the booleans indicating encoding was found."""
+        super().__init__(*args, **kwargs)
+        self._encoding = False
+
+    def visit_Call(self, node: ast.Call):
+        """Visit calls and finds if it is an open function."""
+        if isinstance(node.func, ast.Name) and node.func.id == 'open':
+            if node.keywords:
+                for keyword in node.keywords:
+                    self._check_keywords(keyword)
+            if node.args:
+                self._check_args(node.args)
+
+            if self._encoding is False:
```

Comment:
`if not is_positional and not is_keyword:`
And that's how we can remove the global state.